### DISCUSSION TITLE
Why refresh the page, it will run into the detail content in react/next.js project?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [vercel/next.js#49637] [Help] [answered]
### Summary

I have a electron project base on react/next.js with react-router-dom.

all works well, but there is a issue:
when I refresh the page, it will run into the detail content,
check those snapshots:

[![enter image description here][1]][1] 

[![enter image description here][2]][2]

I have upload this test project to [codesandbox](https://codesandbox.io/s/2svj8m):
if want to run it, should download it (File -> Export to File) , because it not a web project.

  [1]: https://i.stack.imgur.com/CK4YY.png
  [2]: https://i.stack.imgur.com/QFrA9.jpg

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Well I guess the problem is the tech stack... The details page is the `page 2` right?

The problem is that:

- Next.js, pages dir, uses a file based route system, where every file in under the pages directory is a route - this makes Next.js, effectively a multi-entry point SPA. 
- With React Router DOM typically, you want to have a single entry point, often it is just `/`
- Both route systems expect to be in control of navigation

So when you SPA transition to any route, using react-router-dom, you hide it from Next.js as well, but that's not the core issue. The core issue is that, once you reload/refresh the route, you are no longer at `/`, now the Next.js router in the server takes over, and loads the contents of `pages/route/that/matches`

This goes a bit into why we gotta use the router's Link component, and not a simple `a`, because when routers were made, and still today, there's no navigation event, so you literally have to use their Link component, to tell them about a route change. There's history changes event, but nothing for like, actual link navigation. There's an upcoming feature: https://developer.mozilla.org/en-US/docs/Web/API/Navigation/navigate_event, but it is not widely supported yet.

Just be clear, **even that feature wouldn't fix your issue**, the problem is that upon realoding, **the Next.js file based route system takes over, and just sends whatever is in `pages/some/path`, in your case `pages/next/page2`** 

**_Solution: Use Next.js' router, uninstall react-router-dom_** 

> For completeness sake, ome people manage to hack this, and get it to work, but I would NOT recommend it - I did help someone who did the same not long ago: https://github.com/vercel/next.js/discussions/48567#discussion-5105195 they used the `next export` feature though
